src/Microsoft.Identity.Client/Platforms/uap/UapPlatformProxy.cs

```diff
@@ -142,7 +142,7 @@ public override string GetBrokerOrRedirectUri(Uri redirectUri)
         /// <inheritdoc />
         public override string GetDefaultRedirectUri(string correlationId)
         {
-            return Constants.DefaultRedirectUri;
+            return Constants.UapWEBRedirectUri;
```

Comment:
This is breaking change ... Do we really want to force people to use the SSO stuff by default? It would mean that they have to register a specific UAP style redirect URI no? 

CC @jmprieur

Reply:
They would only have to register the UAP specific redirectUri if they are not using the default, or passing in null. Also, the default we use will be changing soon anyway. Is it possible to have two defaults? 

---
In reply to: [288942820](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1182#discussion_r288942820) [](ancestors = 288942820)

Reply:
So if they don't specify a redirect URI (or use null, which is semantically the same), then the interactive flow will fail with a message saying "the redirect uri "uap://foobar123455" is not registered". 

Not a horrible failure, it's kind of in your face, so they'll go and register it. But a breaking change nonetheless.  

It's also a bit inconsistent with the way we handle a similar siutation with ADAL + broker (and soon MSAL + broker?). MSAL knows the redirect URI that needs to be used and if the user chooses smth elese, we throw a meaningful exception. 
In UWP, we choose the redirect URI and let ESTS throw a meaningful exception.

Reply:
sure...I get it. let me think on how to make the customer experience better on this.

Reply:
Regarding the inconsistency in behavior, ADAL + Broker on android can calculate the redirect URI from the signature of the package. This works because there can only be 1 redirect URI for the scenario to work since it is based on the signature. Is it possible for UWP + broker to to use more than one redirect URI? If so, it may not be possible us to throw an exception for the wrong redirect URI. If this is the case, then I dont think we can get around that inconsistency.